cluster/src/dunit/scala/io/snappydata/externalstore/ColumnTableDUnitTest.scala

```diff
@@ -685,6 +685,38 @@ class ColumnTableDUnitTest(s: String) extends ClusterManagerTestBase(s) {
     FileUtils.deleteDirectory(new File(tempPath))
   }
 
+  def testSNAP2088(): Unit = {
+    val snc = org.apache.spark.sql.SnappyContext(sc)
+    val t1 = "snap2088"
+    val t2 = "snap2088_2"
+
+    snc.sql(s"create table $t1 (airport_id int, name string, city string, country string) " +
+        s"using column options (COLUMN_BATCH_SIZE '50')")
+    snc.sql(s"create table $t2 (airport_id int, name string, city string, country string) " +
+        s"using column options (COLUMN_BATCH_SIZE '5000')")
+
+    1 to 500 foreach { i =>
+      if (i % 2 == 0) {
+        snc.sql(s"insert into $t1 values (${Random.nextInt}, 'name_$i', null, 'country_$i')")
+        snc.sql(s"insert into $t2 values (${Random.nextInt}, 'name_$i', null, 'country_$i')")
+      } else {
+        snc.sql(s"insert into $t1 values (${Random.nextInt}, 'name_$i', 'city_$i', 'country_$i')")
+        snc.sql(s"insert into $t2 values (${Random.nextInt}, 'name_$i', 'city_$i', 'country_$i')")
+      }
+    }
```

Comment:
Better to do this in one insert with Spark APIs which also allows inserting much larger set of rows in a reasonable time: `snc.range(10000).selectExpr("cast ((rand() * 100000) as int)", "case when id % 2 = 0 then null else concat("city_", cast(id % 10) as string)").insertInto...`